### PR TITLE
Signup phone number screen improvements

### PR DESCRIPTION
**Pull request by pzduniak**

### CODE REVIEW

**Comment by pzduniak** [comment]
image
image
image
image
image
image
image

**Comment by pzduniak** [comment]
image
replaced spaces in country codes with non-breaking spaces

**Comment by pzduniak** [comment]
Design question: just to make sure, we're enabling phone number visibility by default, right?

File: shared/signup/phone-number/phone-input.tsx [hunk 1]
```diff
@@ -25,13 +33,100 @@ const menuItems = memoize((countryData, onClick) =>
...
+    selected: this.props.selected,
+  }
+
+  componentDidUpdate(prevProps: CountrySelectorProps) {
+    if (this.props.selected !== prevProps.selected) {
+      this._onSelect(this.props.selected)
+    }
+  }
+
+  _onSelect = selected => this.setState(s => (s.selected === selected ? null : {...s, selected}))
```
Comment (buoyad) [inline shared/signup/phone-number/phone-input.tsx]: Don't spread `...s` in here, react only replaces supplied keys

File: shared/util/phone-numbers/sms-support/update-data.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
```
Comment (buoyad) [inline shared/util/phone-numbers/sms-support/update-data.sh]: can you document this somewhere?

File: shared/signup/phone-number/phone-input.tsx (hunk 1 above)
Comment (pzduniak) [reply shared/signup/phone-number/phone-input.tsx]: TIL!

**Pull request merged**